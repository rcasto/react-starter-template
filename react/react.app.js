import { bootStrap as AngularBootStrap } from './bootstrap/angularAppBootstrap';
import { bootStrap as ReactBootStrap } from './bootstrap/reactAppBootstrap';

// Setup a separate react app on the page
ReactBootStrap();

// Hook up to angular app
AngularBootStrap();