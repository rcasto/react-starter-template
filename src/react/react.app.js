import { createElement } from 'react';
import { render } from 'react-dom';
import TestComponent from './components/testComponent';

// Setup a separate react app on the page
const domContainer = document.querySelector('.react-container');
render(createElement(TestComponent, {
    source: 'react'
}), domContainer);