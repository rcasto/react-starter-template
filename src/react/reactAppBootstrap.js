import { createElement } from 'react';
import { render } from 'react-dom';
import TestComponent from '../components/testComponent';

// Setup a separate react app on the page
export function bootStrap(selector) {
    const domContainer = document.querySelector(selector);
    render(createElement(TestComponent, {
        source: 'react'
    }), domContainer);
}