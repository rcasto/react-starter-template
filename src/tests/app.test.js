import { createElement } from 'react';
import renderer from 'react-test-renderer';
import App from '../components/app/app';
import StockCards from '../components/stockCards/stockCards';

const mockStockCardsComponent = createElement('div', {
    className: 'stock-cards-test'
}, 'Stock Cards Component');

jest.mock('../components/stockCards/stockCards', () => {
    return jest.fn().mockImplementation(() => {
        return {
            render: () => mockStockCardsComponent
        };
    });
});

beforeEach(() => {
    StockCards.mockClear();
});

test('Can initialize application component', () => {
    const symbols = ['MSFT', 'AAPL'];
    const component = createElement(App, {
        symbols
    });
    const renderedComponent = renderer.create(component);
    let tree = renderedComponent.toJSON();

    expect(tree).toMatchSnapshot();
    expect(StockCards.mock.calls.length).toEqual(1);
    expect(StockCards).toHaveBeenCalledWith(expect.objectContaining({
        symbols
    }), {});
});