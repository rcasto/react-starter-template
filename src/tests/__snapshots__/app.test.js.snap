// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can initialize application component 1`] = `
<div
  className="app"
>
  <div
    className="stock-cards-test"
  >
    Stock Cards Component
  </div>
</div>
`;
